### PR TITLE
PAYARA-2548 OpenAPI 1.0 Implementation

### PR DESCRIPTION
**Pull request by MattGill98**

### CODE REVIEW

**Comment by MattGill98** [comment]
jenkins test please

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                if (map.keySet().toArray()[0].equals(appInfo)) {
+                    models.remove(map);
+                    break;
+                }
+            }
+        }
+    }
+
+    private String getContextRoot(ApplicationInfo appInfo) {
+        return appInfo.getMetaData(WebBundleDescriptorImpl.class).getContextRoot();
```
Comment (smillidge) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java]: You can get a NPE here for example when deploying an EJB jar as getMetaData I believe can return null.

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/admin/SetOpenApiConfigurationCommand.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+@PerLookup
+@ExecuteOn({ RuntimeType.DAS })
+@TargetType({ DAS, DEPLOYMENT_GROUP, STANDALONE_INSTANCE, CLUSTER, CLUSTERED_INSTANCE, CONFIG })
+@RestEndpoints({
+    @RestEndpoint(configBean = OpenApiServiceConfiguration.class,
+            opType = POST,
+            path = "set-openapi-configuration",
+            description = "Sets the OpenAPI Configuration")
+})
+public class SetOpenApiConfigurationCommand implements AdminCommand {
```
Comment (smillidge) [inline appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/admin/SetOpenApiConfigurationCommand.java]: This command as written will require a restart to take effect. Can you make sure it indicates that in the administration console.

Comment (MattGill98) [reply appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/admin/SetOpenApiConfigurationCommand.java]: How do you mean? From my testing, enabling and disabling the OpenAPI seems dynamic. I've moved the logging into the OpenAPIService to make it more obvious and readable.

File: appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java (hunk 1 above)
Comment (MattGill98) [reply appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenApiService.java]: No worries, I've just made a fix for that.

**Comment by arjantijms** [comment]
Jenkins test

**Comment by MattGill98** [comment]
Jenkins test please

**Comment by MattGill98** [comment]
Jenkins test please

**Pull request merged**